recipe/meta.yaml

```diff
@@ -1,105 +1,117 @@
+        # avoid the solver getting stuck in no-man's-land.
+        - __osx >={{ MACOSX_DEPLOYMENT_TARGET }}  # [osx and x86_64]
+      missing_dso_whitelist:  # [s390x]
+        - $RPATH/ld64.so.1    # [s390x]
     requirements:
       build:
         - {{ compiler('c') }}
         - {{ compiler('cxx') }}
-        - m2-patch  # [win]
-        - patch  # [not win]
-        - cmake >=3.5.1
-        - pkg-config  # [unix]
-        # `protoc` is also used for building
-        - libprotobuf {{ libprotobuf }}
-        - ninja
+        - cmake
+        - ninja-base
+        # During cross-compilation, we need to build the grpc_cpp_plugin for the
+        # architecture of the build agent (where it will be executed afterwards),
+        # thus we need the respective host dependencies also in the build env.
+        - c-ares        # [build_platform != target_platform]
+        - libabseil     # [build_platform != target_platform]
+        - libprotobuf   # [build_platform != target_platform]
+        - openssl       # [build_platform != target_platform]
+        - re2           # [build_platform != target_platform]
+        - zlib          # [build_platform != target_platform]
       host:
-        - abseil-cpp 20230802.0
+        - libabseil 20240116.2
         - c-ares 1.19.1
-        - libprotobuf {{ libprotobuf }}
+        - libprotobuf 4.25.3
         - re2 2022.04.01
         - openssl {{ openssl }}
         - zlib {{ zlib }}
       run:
-        # Need to pin libprotobuf to the same version used when building,
-        # otherwise we may get a different SO ABI version.
-        # For example 3.6.1 is ABI-incompatible with 3.6.0...
-        - {{ pin_compatible("libprotobuf", min_pin="x.x.x", max_pin="x.x.x") }}
-        - zlib >=1.2.13,<1.3.0a0
-        - re2
-        - openssl 3.*
-
-    # core package & vendored libs use different version scheme than CPP libs, see
-    # https://github.com/grpc/grpc/blob/v1.48.1/CMakeLists.txt#L28-L32
-    {% set core_version = "26.0.0" %}
-
-    {% set core_libs = ["gpr", "grpc", "grpc_unsecure"] %}
-    {% set core_cpp_libs = ["grpc++", "grpc++_unsecure"] %}
-    {% set vendored_libs = ["address_sorting", "upb"] %}
-    {% set binaries_plugin_langs = [
-        "cpp", "csharp", "node", "objective_c", "php", "python", "ruby"
-    ] %}
+        - __osx >={{ MACOSX_DEPLOYMENT_TARGET|default("10.9") }}  # [osx and x86_64]
+      run_constrained:
+        # make sure we don't co-install with old version of previous package name
+        - grpc-cpp ={{ version }}
     test:
       requires:
         # protoc needs compiler; cmake also needs one to run package detection, see
         # https://discourse.cmake.org/t/questions-about-find-package-cli-msvc/6194
         - {{ compiler('c') }}
         - {{ compiler('cxx') }}
         - cmake
-        - ninja
+        - ninja-base
         - pkg-config
       files:
-        - tests/include/*
+        - cmake_test/
```

Comment:
I'm failing to see the difference between `files` and `source_files`? Shouldn't `hello.proto` be part of `source_files`?

Reply:
files takes files from the recipe directory: https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#test-files